hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/DBStore.java

```diff
@@ -44,17 +45,20 @@
    */
   Table<byte[], byte[]> getTable(String name) throws IOException;
 
+
   /**
    * Gets an existing TableStore with implicit key/value conversion.
    *
    * @param name - Name of the TableStore to get
    * @param keyType
    * @param valueType
+   * @param cachetype - Type of cache need to be used for this table.
    * @return - TableStore.
    * @throws IOException on Failure
    */
   <KEY, VALUE> Table<KEY, VALUE> getTable(String name,
-      Class<KEY> keyType, Class<VALUE> valueType) throws IOException;
+      Class<KEY> keyType, Class<VALUE> valueType,
+      TableCache.CACHETYPE cachetype) throws IOException;
```

Comment:
Why do we need an external visible TableCache.CACHETYPE ? shouldn't this be an implementation detail of the Tables that have Cache?

Reply:
Added this because for a few tables like bucket and volume table plan is to maintain full table information, for other tables we maintain a partial cache, whereas for few tables we don't want to maintain cache at all. (This is a common interface for all tables in Ozone SCM/OM. So, having this option will help to know which kind of cache need to be used for the table.)

As these are frequently used for validation of almost every operation in OM. So, this might improve validation like bucket/volume exists or not checks.

Reply:
Removed the cache type.